### PR TITLE
Valohai integration

### PR DESCRIPTION
**Pull request by JuhaKiili**
Huggingface / Transformers Valohai integration

Changes to existing Transformers code:
- Prints Valohai-styled logs (JSON)

Additional info:
- valohai.yaml has most (but not all) parameters used by run_glue.py
- Valohai execution downloads all glue datas by default (still pretty fast). Download script placed in `utils/download_glue_data.py`.
- Valohai execution only saves model/checkpoint at the end by default (adjust with Valohai UI)
- Valohai execution logs every 25 steps (adjust with Valohai UI)

### CODE REVIEW

**Comment by akx** [comment]
> * Loss logging no longer accumulates (why was that?) but shows raw loss

With the revised commits, this is no longer true (but I'm not able to edit @JuhaKiili's original message, and he's AFK at Slush :) )

File: examples/run_glue.py [hunk 1]
```diff
@@ -171,13 +172,21 @@ def train(args, train_dataset, model, tokenizer):
...
                         results = evaluate(args, model, tokenizer)
                         for key, value in results.items():
-                            tb_writer.add_scalar('eval_{}'.format(key), value, global_step)
-                    tb_writer.add_scalar('lr', scheduler.get_lr()[0], global_step)
-                    tb_writer.add_scalar('loss', (tr_loss - logging_loss)/args.logging_steps, global_step)
+                            eval_key = 'eval_{}'.format(key)
+                            tb_writer.add_scalar(eval_key, value, global_step)
+                            logs[eval_key] = str(value)
                     logging_loss = tr_loss
+                    loss_scalar = (tr_loss - logging_loss) / args.logging_steps
```
Comment (thomwolf) [inline examples/run_glue.py]: This looks a bit strange to me, `logging_loss = tr_loss` so `loss_scalar = (tr_loss - logging_loss) / args.logging_steps` will always be zero, no?

File: examples/run_glue.py [hunk 2]
```diff
@@ -171,13 +172,21 @@ def train(args, train_dataset, model, tokenizer):
...
+                    learning_rate_scalar = scheduler.get_lr()[0]
+                    tb_writer.add_scalar('lr', learning_rate_scalar, global_step)
+                    tb_writer.add_scalar('loss', loss_scalar, global_step)
+                    logs['learning_rate'] = learning_rate_scalar
+                    logs['loss'] = loss_scalar
```
Comment (thomwolf) [inline examples/run_glue.py]: maybe simpler to add all the metrics in `log` like you do and then just put them all in `tb_writer` at once at the end to avoid all these duplicated lines.

File: utils/download_glue_data.py [hunk 3]
```diff
@@ -0,0 +1,141 @@
+''' Script for downloading all GLUE data.
```
Comment (thomwolf) [inline utils/download_glue_data.py]: We need to add a link to the original location of this script.
And copyright notice as well.

**Review by thomwolf — COMMENTED** [review]
Ok, a few things to update in the scripts as indicated in the comments.
Tell me if you want me to do it.
And then we should add a link in the readme pointing to some explanation on how to run the examples.
One location could be [here](./examples/README.md)

File: examples/run_glue.py (hunk 1 above)
Comment (JuhaKiili) [reply examples/run_glue.py]: Fix pushed

File: examples/run_glue.py (hunk 2 above)
Comment (JuhaKiili) [reply examples/run_glue.py]: Fix pushed

File: utils/download_glue_data.py (hunk 3 above)
Comment (JuhaKiili) [reply utils/download_glue_data.py]: Fix pushed

File: utils/download_glue_data.py [hunk 4]
```diff
@@ -0,0 +1,144 @@
...
+
+Original source: https://github.com/kamalkraj/ALBERT-TF2.0/blob/fa90194e5fe729dbb19f32ac29c8d6d6372c0f93/download_glue_data.py
+Original license: https://github.com/kamalkraj/ALBERT-TF2.0/blob/fa90194e5fe729dbb19f32ac29c8d6d6372c0f93/LICENSE (Apache-2.0)
```
Comment (thomwolf) [inline utils/download_glue_data.py]: That's not the original source, this script was made by NYU.
It should probably be: https://gist.github.com/W4ngatang/60c2bdb54d156a41194446737ce03e2e

Comment (JuhaKiili) [reply utils/download_glue_data.py]: I changed the link to what you suggested, but not sure what is the copyright or the license of a gist? Maybe just the link to the gist is enough?
